Source/Renderer/modernizeShader.js

```diff
@@ -0,0 +1,233 @@
+define([
+        '../Core/defined',
+        '../Core/DeveloperError',
+    ], function(
+        defined,
+        DeveloperError) {
+    'use strict';
+
+    // Note that this fails if your string looks like
+    // searchString[singleCharacter]searchString
+    function replaceInSourceString(str, replacement, splitSource) {
+        var regexStr = '(^|[^\\w])(' + str + ')($|[^\\w])';
+        var regex = new RegExp(regexStr, 'g');
+
+        for (var number = 0; number < splitSource.length; ++number) {
+            var line = splitSource[number];
+            splitSource[number] = line.replace(regex, '$1' + replacement + '$3');
+        }
+    }
+
+    function replaceInSourceRegex(regex, replacement, splitSource) {
+        for (var number = 0; number < splitSource.length; ++number) {
+            var line = splitSource[number];
+            splitSource[number] = line.replace(regex, replacement);
```

Comment:
Yeah, I think the crux of it is that I wanted to deal with line numbers rather than character offsets into the source